q2_demux/plugin_setup.py

```diff
@@ -228,4 +230,43 @@
                  'sequences after subsampling.')
 )
 
+T = TypeMatch([SequencesWithQuality, PairedEndSequencesWithQuality,
+               JoinedSequencesWithQuality])
+plugin.methods.register_function(
+    function=q2_demux.filter_samples,
+    inputs={'demux': SampleData[T]},
+    parameters={'metadata': Metadata,
+                'where': Str,
+                'exclude_ids': Bool},
+    outputs=[
+        ('filtered_samples', SampleData[T])
+    ],
+    input_descriptions={
+        'demux': 'The demultiplexed data from which samples should be '
+        'filtered.'
+    },
+    parameter_descriptions={
+        'metadata': 'Sample metadata indicating which sample ids to filter. '
+                    'The optional `where` parameter may be used to filter ids '
+                    'based on specified conditions in the metadata. The '
+                    'optional `exclude_ids` parameter may be used to exclude '
+                    'the ids specified in the metadata from the filter.',
+        'where': 'Optional SQLite WHERE clause specifying sample metadata '
+                 'criteria that must be met to be included in the filtered '
+                 'data. If not provided, all samples in `metadata` that are '
+                 'also in the demultiplexed data will be retained.',
+        'exclude_ids': 'Defaults to False. If True, the samples selected by '
+                       'the `metadata` and optional `where` parameter will be '
+                       'excluded from the filtered data.',
+    },
+    output_descriptions={
+        'filtered_samples': 'The samples filtered out by the metadata.'
```

Comment:
Update key as above, and make the description something like `The filtered demultiplexed data.`